src/hotspot/share/code/codeBlob.cpp

```diff
@@ -394,6 +394,7 @@ BufferBlob::BufferBlob(const char* name, CodeBlobKind kind, int size)
 {}
 
 BufferBlob* BufferBlob::create(const char* name, uint buffer_size) {
+  MACOS_AARCH64_ONLY(os::thread_wx_enable_write());
```

Comment:
I'm not sure why write has to be enabled here when it is not needed in any of the other create methods. The new operation below will call CodeCache::allocate (just as happens with with, say, VtableBlob::create). So, why is that not enough?